satpy/scene.py

```diff
@@ -1076,12 +1076,16 @@ def to_hvplot(self, datasets=None, *args, **kwargs):
 
 
 
-    def to_xarray_dataset(self, datasets=None):
+    def to_xarray_dataset(self, datasets=None, compat="minimal"):
         """Merge all xr.DataArrays of a scene to a xr.DataSet.
 
         Parameters:
             datasets (list):
                 List of products to include in the :class:`xarray.Dataset`
+            compat (Optional[str]):
```

Comment:
The `Optional` type suggests this can be `None`, but that's not true is it?

Reply:
Hah I think you are right I always thought `optional` does not mean it can be `None` rather if you do not provide it the default arg is used. Then I need to change it. I guess if one uses `optional` you have to check for none inside the function and then set the default. I think that's a little overhead and not very nice but is in accordance with the documentation of `Optional`.

Reply:
Just make it "str" only and have the default be a string. No need for Optional or None checking.